### PR TITLE
Fixes #24699 - Alias reports for run destribution join

### PR DESCRIPTION
**Pull request by bastilian**
This avoids conflicting with other joins and should fix the issue mentioned in the issue.

### CODE REVIEW

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -184,7 +184,8 @@ class Jail < ::Safemode::Jail
     if fromtime.nil? || totime.nil?
       raise ::Foreman.Exception.new(N_("invalid time range"))
     else
-      joins("INNER JOIN reports ON reports.host_id = hosts.id").where("reports.reported_at BETWEEN ? AND ?", fromtime, totime)
+      joins("INNER JOIN reports as run_distribution_reports ON run_distribution_reports.host_id = hosts.id")
```
Comment (tbrisker) [inline app/models/host/managed.rb]: hmm, while this makes sense, why not just use `.joins(:all_reports)` and let rails take care of the naming?

**Review by tbrisker — COMMENTED** [review]
I'm not able to reproduce either, not sure what changed, but see comment inline.

**Comment by tbrisker** [comment]
Since we can't reproduce, it seems like this issue has been resolved elsewhere, closing for now.

File: app/models/host/managed.rb (hunk 1 above)
Comment (guillaume-baffoin) [reply app/models/host/managed.rb]: just for information we have encounter this bug in production with the 18.3 version and  this fix have worked for us.

**Pull request closed without merging**